src/main/java/uk/gov/hmcts/probate/services/persistence/services/SequenceNumberService.java

```diff
@@ -1,25 +1,94 @@
 package uk.gov.hmcts.probate.services.persistence.services;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import javax.persistence.EntityManager;
+import javax.persistence.LockModeType;
+import javax.transaction.Transactional;
+
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.context.config.ResourceNotFoundException;
 import org.springframework.jdbc.support.incrementer.PostgreSQLSequenceMaxValueIncrementer;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.probate.services.persistence.model.Registry;
 import uk.gov.hmcts.probate.services.persistence.model.RegistryNotConfiguredException;
+import uk.gov.hmcts.probate.services.persistence.repository.RegistryRepository;
 
+import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 @Service
+@Transactional
 public class SequenceNumberService {
```

Comment:
This might take a little refactor (SORRY!) but this class is no longer a generic 'sequenceNumberService' but is now a 'registrySequenceService' or something named more akin for identifying the next registry in the sequence.